### ISSUE TITLE
Crush immidiately when disconnect displays anytime after load localhome

### ISSUE THREAD

**Issue by kazu0617** [Yellow-Dog-Man/Resonite-Issues#727]
### Describe the bug?

When any display connect or disconnect after load localhome, resonite crushes immidiately.

### To Reproduce

1. Connect multi display (at least 2),
1. Launch Resonite.
2. Wait until load localhome.
3. disconnect display (if you use DP, just turn off that display).
4. Crush.

### Expected behavior

When I disconnect display, resonite doesn't crush.

### Resonite Version Number

2023.11.15.1158@2019.4.19f1

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Anything. I test in VR(ALVR / Varjo / Vive)/Desktop

### Log Files

[HINATA-WIN - 2023.11.15.1158 - 2023-11-18 20_00_54.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13400084/HINATA-WIN.-.2023.11.15.1158.-.2023-11-18.20_00_54.log)

And here's crush files without dump. it seems this bug curse with "uDesktopDuplication.dll".
[error.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13400087/error.log)
[Player.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13400088/Player.log)

### Additional Context

This crush doesn’t show when disable desktop behavior, so i disable desktop function when i use screen mode.
[Config-DisableDesktop.json](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13400076/Config-DisableDesktop.json)

### Reporters

Resonite: kazu / litalita
Discord:  `@kazu0617` / `@litalita`  ( `<@190137850698792961>` / `<@662107702633168916>` )

**Comment by 989onan** [comment]
I can confirm this works as someone with two monitors

**Comment by Ardescar** [comment]
This is related to this report. https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/79

**Comment by Ardescar** [comment]
For me it only crashes when i remove a display.
Adding one did sofar no crash.

**Comment by kazu0617** [comment]
@Ardescar 
> For me it only crashes when i remove a display. Adding one did sofar no crash.
> 
> But it happens sometimes and by accident that i crash resonite by switching my montor off.

If you turn off the monitor, the crash is probably a mechanism that detects the on / off of the monitor in the display port and switches the connection. In a sense, it is normal, and resonite can cause crashes.

**Comment by JackTheFoxOtter** [comment]
This was the reason I had to disable my main DP monitor's auto-sleep, since everytime it went to sleep, Resonite would crash.

**Comment by kazu0617** [comment]
@JackTheFoxOtter
> This was the reason I had to disable my main DP monitor's auto-sleep, since everytime it went to sleep, Resonite would crash.

By loading config written in this issue, it will not crash without changing the display port settings. Resonite makes the desktop invisible, so it's a temporary measure, but you can check the desktop with other tools, so let's get this hard.

**Comment by JackTheFoxOtter** [comment]
> @JackTheFoxOtter
> 
> > This was the reason I had to disable my main DP monitor's auto-sleep, since everytime it went to sleep, Resonite would crash.
> 
> By loading config written in this issue, it will not crash without changing the display port settings. Resonite makes the desktop invisible, so it's a temporary measure, but you can check the desktop with other tools, so let's get this hard.

I'm aware, however disabling the in-game Desktop functionality isn't an option for me, as I use it a lot. Disabling auto-sleep of my monitor was the better option in my case, even if it results in unnecessary power draw.

**Comment by 3x1t-5tyl3** [comment]
Yup. This is still a thing. Any display that connects or disconnects causes an immediate unity crash. - Beta 2024.12.2.1188

**Comment by Merith-TK** [comment]
Confirmed still a thing. Makes it a pain for remote dev when working on my resolink program because  I dont have often enough time to be at home working on it so when I remote in with RDP it changes the "monitors" and then *crashes*

**Comment by Frooxius** [comment]
We are using open source library for this: https://github.com/hecomi/uDesktopDuplication

I'm pretty sure this needs to be fixed on this library's end.

If anyone wants to have a look into this, that could help! If there's a fixed version (or maybe they already fixed it and we have old version?) we can pull that in.
